### DISCUSSION TITLE
Version 0.9.0 RC1

### DISCUSSION THREAD

**Discussion by s-allius** [s-allius/tsun-gen3-proxy#114] [Announcements]
The version 0.9.0 has the focus on improving stability with the following features:

- `Healthcheck` which observe the message processing
- Readiness and liveness checks for `Kubernetes`
- Synchronise regular MODBUS commands with the inverter's state to avoid that the inverter get unexpected packets
- GEN3PLUS: dump invalid packages with wrong start or stop byte, for easier analyses

Improved error and connection handling:
 - Bugfix GEN3: Not answered by the TSUN Cloud get_time request no longer blocks the connection establishment of the inverter
 - Bugfix: Pending MODBUS requests no longer block the release of connection structures when the connection is terminated
 - Bugfix: Inverter connection no longer closes if there are problems with the connection to the TSUN cloud
 - Detect and disconnect dead connection which received no packets for a long time
 - Bugfix: MODBUS responses for MODBUS requests from the TSUN cloud aren't dropped anymore
 - Bugfix GEN3: New data types of DSP version 5.0.17 added so that they are no longer considered incorrect

## What's Changed
* S allius/issue56 by @s-allius in https://github.com/s-allius/tsun-gen3-proxy/pull/90
* S allius/issue94 by @s-allius in https://github.com/s-allius/tsun-gen3-proxy/pull/96
* S allius/issue93 by @s-allius in https://github.com/s-allius/tsun-gen3-proxy/pull/97
* Health check by @s-allius in https://github.com/s-allius/tsun-gen3-proxy/pull/99
* S allius/issue100 by @s-allius in https://github.com/s-allius/tsun-gen3-proxy/pull/101
* S allius/issue108 by @s-allius in https://github.com/s-allius/tsun-gen3-proxy/pull/109
* S allius/issue102 by @s-allius in https://github.com/s-allius/tsun-gen3-proxy/pull/110
* S allius/issue111 by @s-allius in https://github.com/s-allius/tsun-gen3-proxy/pull/112

**Full Changelog**: https://github.com/s-allius/tsun-gen3-proxy/compare/v0.8.1...v0.9.0-rc.1

<hr /><em>This discussion was created from the release <a href='https://github.com/s-allius/tsun-gen3-proxy/releases/tag/v0.9.0-rc.1'>Version 0.9.0 RC1</a>.</em>

**Comment by Cardes** [comment]
I tested this release in kubernetes with healtchecks, everything works fine.
Thanks for your effort!

**Reply by Ivan-Dobynda** [reply]
Me too
